src/receive_imf.rs

```diff
@@ -2324,6 +2329,82 @@ async fn handle_edit_delete(
     Ok(())
 }
 
+async fn handle_post_message(
+    context: &Context,
+    mime_parser: &MimeMessage,
+    from_id: ContactId,
+) -> Result<()> {
+    if let Some(mimeparser::PreMessageMode::PostMessage) = &mime_parser.pre_message {
+        // if Pre-Message exist, replace attachment
+        // only replacing attachment ensures that doesn't overwrite the text if it was edited before.
+        let rfc724_mid = mime_parser
+            .get_rfc724_mid()
+            .context("expected Post-Message to have a message id")?;
+
+        let Some(msg_id) = message::rfc724_mid_exists(context, &rfc724_mid).await? else {
+            warn!(
+                context,
+                "Download Post-Message: Database entry does not exist."
+            );
+            return Ok(());
+        };
+        let Some(original_msg) = Message::load_from_db_optional(context, msg_id).await? else {
+            // else: message is processed like a normal message
+            warn!(
+                context,
+                "Download Post-Message: pre message was not downloaded, yet so treat as normal message"
+            );
+            return Ok(());
+        };
+
+        if original_msg.from_id != from_id {
+            warn!(context, "Download Post-Message: Bad sender.");
+            return Ok(());
```

Comment:
So this also will be processed like a normal message (looking at the comment above)? Why not `ensure!` to error out in this case?